packages/wdio-browserstack-service/src/service.ts

```diff
@@ -110,6 +113,97 @@ export default class BrowserstackService implements Services.ServiceInstance {
+                const aiHealConfig: any = tcgConfig
+                return {
+                    sessionToken: tcgConfig.sessionToken,
+                    isAuthenticated : tcgConfig.isAuthenticated,
+                    groupID: tcgConfig.groupId,
+                    userID: tcgConfig.userId,
+                    groupAIEnabled: tcgConfig.groupAIEnabled,
+                    isHealingEnabled: aiHealConfig.isHealingEnabled
+                }
+            } catch (err){
+            //   logger.debug(`Cound not setup tcgAuth config file due to error: ${err}`)
+            }
+
+        }
+
+        //   const isUndefinedOrNull: any = val:any => {return (val === undefined || val === null || val === '')}
+
+        //   const isNotUndefinedOrNull: any = val:any => {return !isUndefinedOrNull(val)}
+
+        //   const isString: any = val:any => {return (isNotUndefinedOrNull(val) && typeof val === 'string')}
+
+        function isUndefinedOrNull(val: any){
+            return (val === undefined || val === null || val === '')
+        }
+
+        function isNotUndefinedOrNull(val: any){
+            return !isUndefinedOrNull(val)
+        }
+
+        function isString(val: any){
+            return (isNotUndefinedOrNull(val) && typeof val === 'string')
+        }
+
+        const authInfo:any = readTcgAuthConfigToGlobal()
+        const isAuthenticated:any = authInfo.isAuthenticated
+        const isHealingEnabled:any = authInfo.isHealingEnabled
+        if (isAuthenticated && isHealingEnabled) {
+            console.log('User is authenticated')
+            const sessionToken: any = authInfo.sessionToken
+            const resp: any = await aiSDK.BrowserstackHealing.setToken(this._browser.sessionId, sessionToken, 'https://tcg.browserstack.com')
+            console.log('Response from setToken: ', resp)
+
+            const browser:any = this._browser as WebdriverIO.Browser
+            browser.overwriteCommand('findElement', async (orginalFunc:any, using: any, value: any) => {
+                // const browser = this._browser as WebdriverIO.Browser
+                const sessionId: any = browser.sessionId
+                console.log('HEALING!!!!, patching find element')
+                const constants:any = {
+                    tcgRegion: 'use',
+                    tcgUrl: 'https://tcg.browserstack.com',
+                }
+                let tcgDetails = `{"region": "${constants.tcgRegion}", "tcgUrls": {"${constants.tcgRegion}": {"endpoint": "${constants.tcgUrl.split('s://')[1]}"}}}`
+                tcgDetails = isNotUndefinedOrNull(tcgDetails) && isString(tcgDetails) ? tcgDetails.replace(/'/g, '\\\'').replace(/"/g, '\\"') : ''
+                const using1: string = isNotUndefinedOrNull(using) && isString(using) ? using.replace(/'/g, '\\\'').replace(/"/g, '\\"') : ''
+                const value1: string = isNotUndefinedOrNull(value) && isString(value) ? value.replace(/'/g, '\\\'').replace(/"/g, '\\"') : ''
```

Comment:
Let's avoid such numbering for variables, if `using` or `value` is already used, have a more concrete name for this assignment.